bootstrap/bootstrap.go

```diff
@@ -368,20 +368,37 @@ func (b *Bootstrap) StartMetricsServer(ctx context.Context) error {
 	b.logger.Info().Msg("bootstrap starting metrics server")
 
 	b.metrics = flowMetrics.NewServer(b.logger, uint(b.config.MetricsPort))
-	err := util.WaitClosed(ctx, b.metrics.Ready())
-	if err != nil {
-		return fmt.Errorf("failed to start metrics server: %w", err)
+
+	// this logic is needed since the metric server is a component.
+	// we need to start and stop it manually here.
+
+	ictx, errCh := irrecoverable.WithSignaler(ctx)
+	b.metrics.Start(ictx)
+	<-b.metrics.Ready()
```

Comment:
if the context is cancelled before the metric server is running, this will block forever, 
```suggestion
	if err := util.WaitClosed(ctx, b.metrics.Ready()); err != nil {
		return fmt.Errorf("failed to start metrics server: %w", err)
	}
```